src/runtime/internal/Component.ts

```diff
@@ -237,12 +237,36 @@ if (typeof HTMLElement === 'function') {
 export class SvelteComponent {
 	$$: T$$;
 	$$set?: ($$props: any) => void;
-
+	/**
+	 * 
+	 * ```ts
+	 * component.$destroy()
+	 * ```
+	 * 
+	 * Removes a component from the DOM and triggers any `onDestroy` handlers.
+	 * 
+	 */
 	$destroy() {
 		destroy_component(this, 1);
 		this.$destroy = noop;
 	}
-
+	/**
+	 * Causes the `callback` function to be called whenever the component dispatches an `event`.
+	 * 
+	 * A function is returned that will remove the event listener when called.
+	 * 
+	 * ```ts
+	 * const off = app.$on('selected', event => {
+	 * 	console.log(event.detail.selection);
+	 * });
+	 * 
+	 * off();
+	 * ```
+	 * 
+	 * @param type 
```

Comment:
I think if we're going to add `@param` tags then they should have descriptions of those parameters